dispatcher/src/components/FilterMobile/FilterPage.tsx

```diff
@@ -11,58 +13,73 @@ import {
   FilterText,
   FilterSelection,
 } from "./StyledFilterMobile";
+import SubFilterPage from "./SubFilterPage";
+import categories from "../../data/categories.json";
 
 export interface FilterPageProps {
   inDetails?: boolean;
 }
 
 const FilterPage: React.FC<FilterPageProps> = ({ inDetails }) => {
+  const filter = useSelector((state: RootState) => state.filter);
+
+  const [lineClick, setLineClick] = useState("");
+  const [backClick, setBackClick] = useState(false);
+
+  const backClickHandler = (isClick: boolean) => {
+    setBackClick(isClick);
+  };
+
   return (
     <FilterMobileContainer>
-      {inDetails ? (
-        <FilterSubText>
-          <FilterBackArrow src={blueArrow} />
-          SOURCES
-        </FilterSubText>
-      ) : (
-        <FilterTitle>FILTER</FilterTitle>
-      )}
-      <FullDivider />
-      {!inDetails && (
+      {!backClick ? (
         <>
+          <FilterTitle>FILTER</FilterTitle>
+          <FullDivider />
           <FilterLine>
             <FilterText>Search in</FilterText>
             <FilterSelection>
               <Dropdown isSearch={true} withArrow={false} />
             </FilterSelection>
           </FilterLine>
           <FullDivider />
+          {filter.mainFilter === "Everything"
```

Comment:
Try using enum or union type to define all kinds of filters in a single place.
(Filters.everything instead of hardcoded "Everything") 
This method will prevent a lot of bugs in the future.